### ISSUE TITLE
Flaky test in gnet/pool

### ISSUE THREAD

**Issue by gz-c** [skycoin/skycoin#1827]
Excerpt:

```
[2018-08-30T03:47:01Z] ERROR [gnet]: daemon.gnet.ConnectionPool.BroadcastMessage error: Connection pool is empty
[2018-08-30T03:47:01Z] ERROR [gnet]: req.Func daemon.gnet.ConnectionPool.BroadcastMessage failed: Connection pool is empty
--- FAIL: TestPoolBroadcastMessage (0.11s)
	Error Trace:	pool_test.go:1075
	Error:      	Should be true
[2018-08-30T03:47:01Z] INFO [gnet]: Listening for connections on 127.0.0.1:50823...
[2018-08-30T03:47:01Z] INFO [gnet]: Connection pool closed
[2018-08-30T03:47:02Z] INFO [gnet]: ConnectionPool.Shutdown called
[2018-08-30T03:47:02Z] INFO [gnet]: ConnectionPool.Shutdown closed pool.quit
[2018-08-30T03:47:02Z] INFO [gnet]: ConnectionPool.Shutdown waiting for strandDone
[2018-08-30T03:47:02Z] INFO [gnet]: ConnectionPool.Shutdown closing the listener
[2018-08-30T03:47:02Z] INFO [gnet]: ConnectionPool.Shutdown disconnecting all connections
[2018-08-30T03:47:02Z] INFO [gnet]: ConnectionPool.Shutdown waiting for done
--- FAIL: TestPoolReceiveMessage (0.20s)
	Error Trace:	pool_test.go:1096
			asm_amd64.s:1333
	Error:      	Received unexpected error:
	            	listen tcp 127.0.0.1:50823: bind: address already in use
FAIL
```

Happening on travis intermittently

**Comment by gz-c** [comment]
The failure in `TestPoolReceiveMessage` is because `TestPoolBroadcastMessage` didn't get to clean up after failure; this is normal, ignore `TestPoolReceiveMessage`.

**Comment by gz-c** [comment]
Possibly fixed by #1828, see if it returns

**Comment by gz-c** [comment]
Has not returned yet
